app/src/main/java/com/android/sample/model/user/UserViewModel.kt

```diff
@@ -192,54 +196,94 @@ class UserViewModel(
-      updateList(_fridge, Pair(ingredient, count), add = true)
-    }
-    updateCurrentUser()
+  private fun addIngredientToUserFridge(
+      ingredient: Ingredient,
+      quantity: String,
+      expirationDate: LocalDate
+  ) {
+    val newFridgeItem = FridgeItem(ingredient.barCode.toString(), quantity, expirationDate)
+    updateList(_fridgeItems, Pair(newFridgeItem, ingredient), add = true)
   }
 
   /**
-   * Removes an ingredient from the user fridge. More precisely, it will decrease the count of said
-   * ingredient in the fridge.
+   * Method that removes an [Ingredient] from the fridge of the user and updates the user in the
+   * database
    *
    * @param ingredient the ingredient to be removed
-   * @param count the number of ingredients to be removed
    * @throws IllegalArgumentException if the count is greater than the number of ingredients in the
    *   fridge
    */
-  fun removeIngredientFromUserFridge(ingredient: Ingredient, count: Int = 1) {
+  fun removeIngredientFromUserFridge(ingredient: Ingredient) {
     try {
-      val changedElement: Pair<Ingredient, Int> =
-          _fridge.value.first { it.first.barCode == ingredient.barCode }
-
-      if (changedElement.second - count > 0) {
-        updateList(_fridge, changedElement, add = false)
-        updateList(_fridge, Pair(changedElement.first, changedElement.second - count), add = true)
-      } else if (changedElement.second - count == 0) {
-        updateList(_fridge, changedElement, add = false)
-        updateList(_fridge, changedElement, add = false)
-      } else {
-        throw IllegalArgumentException(REMOVED_TOO_MANY_INGREDIENTS_ERROR)
-      }
+      val deletedIngredient: Pair<FridgeItem, Ingredient> =
+          _fridgeItems.value.first { it.first.id == ingredient.barCode.toString() }
+      updateList(_fridgeItems, deletedIngredient, add = false)
     } catch (e: NoSuchElementException) {
       throw IllegalArgumentException(REMOVED_INGREDIENT_NOT_IN_FRIDGE_ERROR)
     }
     updateCurrentUser()
   }
 
+  /**
+   * Method that either add an ingredient to the user's fridge if it does not exist, and updates it
+   * with the new quantity if it is already in the fridge
+   *
+   * @param ingredient the ingredient to be updated/added
+   * @param quantity a valid quantity for the ingredient
+   * @param expirationDate the expiration date of the ingredient
+   * @param scannedItem boolean that the determines if we update via a scanned item or not
+   */
+  fun updateIngredientFromFridge(
+      ingredient: Ingredient,
+      quantity: String,
+      expirationDate: LocalDate,
+      scannedItem: Boolean
+  ) {
+    try {
+      val changedIngredient: Pair<FridgeItem, Ingredient> =
+          _fridgeItems.value.first {
+            (it.first.id == ingredient.barCode.toString()) &&
+                (it.first.expirationDate == expirationDate)
+          }
+      if (scannedItem) { // case there we scan an item that we want to add to the fridge
```

Comment:
I find this idea intriguing but also somewhat concerning in its current form. From what I understand, the goal is to handle cases where a user scans the same item multiple times. For example, if someone buys four strawberry yogurts, the system would create a single FridgeItem and update the quantity accordingly.

While I appreciate the practicality of this approach, I see a potential issue: four individual yogurts of 100g each are not the same as a single large pot of 400g. For instance, if the user eats one yogurt daily, it’s much simpler to track and remove one yogurt at a time rather than adjusting a larger quantity manually.

In conclusion, I believe we need to pay extra attention to defining the unit of an item in the fridge. Should it represent the total weight, or should it reflect the number of individual pieces? Each approach has implications for usability and practicality. What do you think?

Reply:
After talking with @AdriB02, I have chosen to make the quantity an `Int` in order to represent the number of individual pieces. This has been chosen because it makes the implementation in the UI much easier. Tell me if you are ok with this decision.

Reply:
Yes!